src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,117 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.NothingCommand;
+import duke.command.TodoCommand;
+import duke.command.UnMarkCommand;
+import duke.exceptions.ChatException;
+
+
+/**
+ * General parser for commands from the user.
+ */
+public class Parser {
+
+    /**
+     * Wrapper around Integer.parseInt to re-raise our
+     * own exception.
+     *
+     * @param val The value to parse to an int.
+     * @return Parsed value.
+     * @throws ChatException if the value is invalid.
+     */
+    private static int parseInt(String val) {
+        try {
+            return Integer.parseInt(val);
+        } catch (NumberFormatException e) {
+            throw new ChatException("Task number incorrect format!");
+        }
+    }
+
+    /**
+     * Simple validator to ensure the user doesn't pass in
+     * bad content.
+     *
+     * @param line      User input.
+     * @param startFrom Which character to start checking from.
+     * @return Processed string.
+     */
+    private static String getContents(String line, int startFrom) {
+        String msg = line.substring(startFrom);
+        if (msg.equals(" ") || msg.isEmpty()) {
+            throw new ChatException("The description of a todo cannot be empty.");
+        }
+        msg = msg.stripLeading();
+        return msg;
+    }
+
+    /**
+     * Parse a date in the format day/Month/Year HoursMinutes
+     *
+     * @param date Raw input to pass.
+     * @return A LocalDateTime object of the parse input.
+     * @throws ChatException if the input is invalid
+     */
+    private static LocalDateTime dateParse(String date) {
+        try {
+            return LocalDateTime.parse(date, DateTimeFormatter.ofPattern("d/MM/yyyy HHmm"));
+        } catch (DateTimeParseException e) {
+            throw new ChatException("Invalid date time format");
+        }
+    }
+
```

Comment:
Minor nit: to comply with the coding standard, this method might be better named `parseDate` instead.

Reply:
That makes sense. Thank you.